### ISSUE TITLE
prevent_destroy is not working

### ISSUE THREAD

**Issue by lambertpan** [gruntwork-io/terragrunt#560]
I am trying to use ` prevent_destroy`,  but it doesn't work for me.

When I apply `terragrunt destroy`,  it shows the resource list, and after I hit yes, it goes for delete.

My terragrunt version is v0.16.8

I have give my code below.  

```
terragrunt = {
  # Terragrunt will copy the Terraform configurations specified by the source parameter, along with any files in the
  # working directory, into a temporary folder, and execute your Terraform commands in that folder.
  terraform {
    source = "###my_module"
  }

  prevent_destroy = true

  # Include all settings from the root terraform.tfvars file
  include = {
    path = "${find_in_parent_folders()}"
  }
}
```

**Comment by brikis98** [comment]
@ozerovandrei Got a min to look into this?

**Comment by ozerovandrei** [comment]
@brikis98 sure, I will check it.

**Comment by tgm4883** [comment]
Not to just +1 this but I'm having the same issue on v0.16.7.

**Comment by ozerovandrei** [comment]
Sorry for the delay.
Unfortunately I can't see into it until next weekends or so.

**Comment by ozerovandrei** [comment]
@lambertpan @tgm4883 I can't reproduce this, maybe my modules or variables structure somehow differs from yours.

Could you please provide full example of your configuration that is being destroyed? I can create some additional infrastructure tests for your cases and will try to fix it.

**Comment by lambertpan** [comment]
I have given my code below; my work case is that I create IAM, since it is global I don't want that get deleted. Thanks for looking into it.
```
terragrunt = {
  # Terragrunt will copy the Terraform configurations specified by the source parameter, along with any files in the
  # working directory, into a temporary folder, and execute your Terraform commands in that folder.
  terraform {
    source = "###my_module"
  }

  prevent_destroy = true

  # Include all settings from the root terraform.tfvars file
  include = {
    path = "${find_in_parent_folders()}"
  }
}
```

**Comment by mbering** [comment]
I updated to v16.0.10 and am still experiencing this issue as well.
I have a folder structure similar to:
- infrastructure
  - iam
  - s3
  - vpc
    - terraform.tfvars
  - variables.tf

The vpc terraform.tfvars is where I have the prevent_destroy.

```
terragrunt = {
  # Terragrunt will copy the Terraform configurations specified by the source parameter, along with any files in the
  # working directory, into a temporary folder, and execute your Terraform commands in that folder.
  terraform {
    source = "git::ssh://git//vpc-peering"

        extra_arguments "custom_vars" {
          commands = ["${get_terraform_commands_that_need_vars()}"]

          arguments = [
           "-var-file=${get_tfvars_dir()}/${find_in_parent_folders("variables.tfvars")}"
         ]
      }
  }

  prevent_destroy = true

  # Include all settings from the root terraform.tfvars file
  include = {
    path = "${find_in_parent_folders("terraform.tfvars")}"
  }
}

```

Command used is:
terragrunt destroy-all --terragrunt-non-interactive --terragrunt-iam-role ROLE --terragrunt-download-dir DIR

**Comment by ozerovandrei** [comment]
Sorry for the delay again, haven't got any free time lately.
It seems that I found an issue, `prevent_destroy` is ignored while merging configs with `include` configuration parameter.

I added new integration test with fix in #579.

**Comment by brikis98** [comment]
Should be fixed in #579 and released in https://github.com/gruntwork-io/terragrunt/releases/tag/v0.16.12. Thanks @ozerovandrei!
